src/utils/power_flow/make_pf.jl

```diff
@@ -41,124 +31,106 @@ res = NLsolve.nlsolve(pf!, x0)
 """
 function make_pf(system)
     buses = sort(collect(get_components(Bus, system)), by = x -> get_number(x))
-    bus_count = length(buses)
-    #assumes the ordering in YBus is the same as in the buses.
+    N_BUS = length(buses)
+
+    # assumes the ordering in YBus is the same as in the buses.
     Yb = Ybus(system)
+    x0 = zeros(N_BUS * 2)
 
-    #internally allocate matrices
-    internal = quote
-        V = Vector{ComplexF64}(undef, $bus_count)
-        Vc = Vector{ComplexF64}(undef, $bus_count)
-        P_bal = Vector{Float64}(undef, $bus_count)
-        Q_bal = Vector{Float64}(undef, $bus_count)
-    end
+    # Use vectors to cache data for closure
+    # These should be read only
+    P_GEN_BUS  = fill(0.0, N_BUS)
+    Q_GEN_BUS  = fill(0.0, N_BUS)
+    P_LOAD_BUS = fill(0.0, N_BUS)
+    Q_LOAD_BUS = fill(0.0, N_BUS)
+
+    BUSES = [b for b in enumerate(buses)]
```

Comment:
There is no need to allocate a matrix here you can do  `BUSES = (b for b in enumerate(buses))` to use a generator or simple do enumerate buses below

Reply:
I didn't think we could use a generator because it is being consumed more than once. But it turns out you can, so I'll change it.

Reply:
Dan showed me some significant improvements in Julia in many cases that we use list comprehensions and we really just need generators.

Reply:
Resolved in be481bb